crates/burn/src/lib.rs

```diff
@@ -96,8 +97,45 @@
 
 pub use burn_core::*;
 
+#[cfg(feature = "std")]
+pub use burn_core::{data, lr_scheduler, optim};
+
+/// Backend module.
+pub mod backend {
+    #[cfg(feature = "autodiff")]
+    pub use burn_core::backend::{autodiff, Autodiff};
+    #[cfg(feature = "candle")]
+    pub use burn_core::backend::{candle, Candle};
+    #[cfg(feature = "cuda-jit")]
+    pub use burn_core::backend::{cuda_jit, CudaJit};
+    #[cfg(feature = "tch")]
+    pub use burn_core::backend::{libtorch, LibTorch};
+    #[cfg(feature = "ndarray")]
+    pub use burn_core::backend::{ndarray, NdArray};
+    #[cfg(feature = "wgpu")]
+    pub use burn_core::backend::{wgpu, Wgpu};
+}
+
 /// Train module
 #[cfg(feature = "train")]
 pub mod train {
     pub use burn_train::*;
+
+    /// The metric module.
+    pub mod metric {
+        pub use burn_train::metric::*;
+
+        #[cfg(feature = "metrics")]
+        pub use burn_train::metric::{
+            CpuMemory, CpuTemperature, CpuUse, CudaMetric, TopKAccuracyInput, TopKAccuracyMetric,
+        };
```

Comment:
I don't like the fact we need to know when adding a new metrics, that it should be manually re-exported in the burn root crate. Only for the feature flag to show on the docs :/